### PR TITLE
Remove useless properties and renovate rules

### PR DESCRIPTION
**Pull request by meziantou**
## Additional checks

- [ ] Updated the documentation of the project to reflect the changes
- [ ] Added new tests that cover the code changes

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -13,18 +13,6 @@
       "extends": [
         ":separateMajorReleases"
       ]
-    },
```
Comment (meziantou) [inline renovate.json:16]: I've updated xunit in the previous PR, so this rule is not needed anymore

File: src/Directory.Build.props [hunk 2]
```diff
@@ -3,20 +3,13 @@
...
-    <PublishRepositoryUrl>true</PublishRepositoryUrl>
     <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
     <LangVersion>10</LangVersion>
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
-    <AnalysisLevel>latest-All</AnalysisLevel>
     <Description>Provides MongoDB access through .NET dependency injection, following Microsoft.Extensions.* library practices.</Description>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
```
Comment (meziantou) [inline src/Directory.Build.props:16]: Remove some properties that are part of the coding standard

File: renovate.json (hunk 1 above)
Comment (Gcaya) [reply renovate.json:16]: Updating xunit worked with the analyzers test package? I think that's the issue we were trying to avoid with this rule?

**Review by Gcaya — APPROVED** [review]

File: renovate.json (hunk 1 above)
Comment (meziantou) [reply renovate.json:16]: That's an issue only when people don't read the doc ;)

https://github.com/dotnet/roslyn-sdk/blob/e9aff578fa3d6d30a1e753dbaa604f87f312ebf0/src/Microsoft.CodeAnalysis.Testing/README.md#obsolete-packages

**Pull request merged**